### PR TITLE
Card states, footer button states, emojis and overall changes 

### PR DESCRIPTION
**Pull request by mbosselmann**
Today we learned a lot about states and I added the feature of showing and hiding detailed information about each HP character with a button. We also experimented with emojis today and my result is also reflected in the code. At the end of the day we worked on the navbar in the footer and in my app it is now possible to see which button is currently active. Filtering by Hogwarts houses will probably follow tomorrow. As always, I'm happy to receive feedback. :-)

### CODE REVIEW

File: src/components/card/Card.js [hunk 1]
```diff
@@ -17,33 +19,106 @@ function tellHouseColor(hogwartsHouse) {
 
 function Card(props) {
   const houseColor = tellHouseColor(props.hogwartsHouse);
+  // const showDetails = true;
```
Comment (macfrei) [inline src/components/card/Card.js:22]: Please remove unused code!

File: src/components/card/Card.js [hunk 2]
```diff
@@ -17,33 +19,106 @@ function tellHouseColor(hogwartsHouse) {
...
+        <button
+          className={!showDetails ? "expand-more-button" : "expand-less-button"}
+          onClick={() => {
+            setShowDetails(!showDetails);
+          }}
+        >
+          <img
+            className="expand-more-img"
+            src={!showDetails ? expandMore : expandLess}
+            alt=""
```
Comment (macfrei) [inline src/components/card/Card.js:46]: Here you actually need an `alt` as this image is essential for the user to understand that you can get more information when clicking it.

File: src/components/card/Card.js [hunk 3]
```diff
@@ -17,33 +19,106 @@ function tellHouseColor(hogwartsHouse) {
...
-          <li>Wood: {props.wandWood}</li>
-          <li>Core: {props.wandCore}</li>
-          <li>Length: {props.wandLength}</li>
-        </ul>
-      </div>
+      {showDetails && (
+        <div className="section-content">
+          <div className="content-text">
+            <ul className="main-info-list">
+              <h3>Basics:</h3>
```
Comment (macfrei) [inline src/components/card/Card.js:54]: Only `li` elements are valid children of lists in HTML! The `h3` should be outside of the `ul`. You can also check out [description lists](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl)

File: src/components/card/Card.js [hunk 4]
```diff
@@ -17,33 +19,106 @@ function tellHouseColor(hogwartsHouse) {
...
+            {props.characterName === "Harry Potter" ? (
+              <div className="emoji">⚡</div>
+            ) : (
+              <div></div>
+            )}
+            {props.characterName === "Hermione Granger" && (
+              <div className="emoji">📚</div>
+            )}
+            {props.characterName === "Ron Weasley" && (
+              <div className="emoji">🕸️</div>
```
Comment (macfrei) [inline src/components/card/Card.js:74]: Hahahaha this cracks me up!

File: src/components/footer/Footer.js [hunk 5]
```diff
@@ -1,13 +1,64 @@
...
   return (
     <footer>
-      <nav id="house-filter">
-        <button className="nav-button-gryffindor">G</button>
-        <button className="nav-button-ravenclaw">R</button>
-        <button className="nav-button-hufflepuff">H</button>
-        <button className="nav-button-slytherin">Allies</button>
+      <nav id="house-filter" aria-label="hogwarts houses menu">
+        <button
+          aria-label="gryffindor"
```
Comment (macfrei) [inline src/components/footer/Footer.js:11]: Nice!

File: src/components/footer/Footer.js [hunk 6]
```diff
@@ -1,13 +1,64 @@
...
+  const [activeHouse, setActiveHouse] = useState("");
+
   return (
     <footer>
-      <nav id="house-filter">
-        <button className="nav-button-gryffindor">G</button>
-        <button className="nav-button-ravenclaw">R</button>
-        <button className="nav-button-hufflepuff">H</button>
-        <button className="nav-button-slytherin">Allies</button>
+      <nav id="house-filter" aria-label="hogwarts houses menu">
```
Comment (macfrei) [inline src/components/footer/Footer.js:9]: Don't know whether you need an `aria-label` on the `nav` 🤔

**Review by macfrei — COMMENTED** [review]
Well done! 👍

File: src/components/card/Card.js (hunk 1 above)
Comment (mbosselmann) [reply src/components/card/Card.js:22]: Done!

File: src/components/card/Card.js (hunk 2 above)
Comment (mbosselmann) [reply src/components/card/Card.js:46]: Done!

File: src/components/card/Card.js (hunk 3 above)
Comment (mbosselmann) [reply src/components/card/Card.js:54]: Ok! Fixed this. :-)

File: src/components/footer/Footer.js (hunk 6 above)
Comment (mbosselmann) [reply src/components/footer/Footer.js:9]: Read this article about navigations and aria-labels: [Link](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Navigation_Role)
According to this article I should use an aria-label for `nav`, or?

**Comment by mbosselmann** [comment]
@macfrei Thank you for your remarks! Have changed all mistakes and you will find the changes in my new pull-request. 🎁

File: src/components/footer/Footer.js (hunk 6 above)
Comment (macfrei) [reply src/components/footer/Footer.js:9]: Oh, I didn't know that! 👍 Looks like you are using it correctly, thank you for sharing!

**Pull request merged**